Fix #1782 Add link to Jobs Dashboard after submitting job

technerdlove opened the pull request:
Fixes #1782

#### Changes proposed in this Pull Request:

* Add a link to the job dashboard after submitting a job. Link appears next to the message "Jobs submitted successfully. Your listing will be visible once approved." 
    * If the job dashboard page doesn't have a title, the link text states "View your Jobs" or whatever name the user gave to "Jobs".
    * If the job dashboard page has a title, the link text is the page title.


#### Testing instructions:

**_Page with title_**
* Make sure you have a page with shortcode `[job_dashboard]`. Make sure page has a title. Make sure page is designated as Job Dashboard Page in WP Admin dashboard -> Job Listings -> Settings -> Pages. Keep page open in separate tab or window.
* Make sure you have a page with shortcode `[submit_job_form]`. Make sure page is designated as Submit Job Form Page in WP Admin dashboard -> Job Listings -> Settings -> Pages.
* Make sure the following settings are activated in WP Admin dashboard -> Job Listings -> Settings -> Job Submission:
    * "Require an account to submit listings"
    * "Enable account creation during submission"
    * "Require admin approval of all new listing submissions"
* View page with shortcode `[submit_job_form]`. Create and submit listing as a non-admin user.
* Note confirmation message.

**_Page without title_**
* Edit page with shortcode `[job_dashboard]`as follows: Delete page title and update page. View page.
* Create and submit listing as a non-admin user.
* Note different confirmation message.

<!-- Add the following only if this is meant to be in changelog -->
#### Proposed changelog entry for your changes:
Add link to job dashboard after submitting job

technerdlove committed:
Add link to Jobs Dashboard after submitting job

Uses title set by user in link to job-dashboard.
If there is no title, provides a sentence and link to view plural name of job listing post type.

technerdlove committed:
Update job-submitted.php

technerdlove committed:
Update job-submitted.php

technerdlove committed:
Update job-submitted.php

technerdlove committed:
Remove database query, revert to previous commit

Commit contained a database query for a page that had the `[job_dashboard]` shortcode.  

This should have been accomplished using `WP_Query(args)` anyway,  but nevertheless the commit is reverted to the previous commit for two reasons:
    1. Consistency
        - The user currently can designate the Job Dashboard Page by going to WP Admin dashboard -> Job Listings -> Settings -> Pages 
        - Underneath the dropdown is the following text, which communicates how the plugin locates their Job Dashboard Page : 
            > Select the page where you've used the [job_dashboard] shortcode. This lets the plugin know the location of the dashboard.
        - Thus, even if they cut and paste the `[job_dashboard]` shortcode onto another page,  
           an expectation has been set for them to return to Pages and use the dropdown to designate the new Job Dashboard Page.
    2. "Readability"
        - Reuse available code, reduce number of lines

jom commented on templates/job-submitted.php:
```diff
@@ -35,6 +35,30 @@
 		);
+
+		$permalink = job_manager_get_permalink('job_dashboard');
+		$title = get_the_title((job_manager_get_page_id('job_dashboard')));
```
PHPCS: Align equal sign with one above with spaces and remove extra parens.

```suggestion
		$title     = get_the_title( job_manager_get_page_id( 'job_dashboard' ) );
```

jom commented on templates/job-submitted.php:
```diff
@@ -35,6 +35,30 @@
+		$title = get_the_title((job_manager_get_page_id('job_dashboard')));
+
+		// If job_dashboard page exists but there is no title
+		if ( $permalink && empty($title) ) {
```
Formatting around parens.

```suggestion
		if ( $permalink && empty( $title ) ) {
```

jom reviewed (changes requested):
Hi there! A few minor formatting issues. Thanks for your PR!

technerdlove committed:
Clean up formatting of var $title

PHPCS: Align equal sign with one above with spaces and remove extra parens

Co-Authored-By: Jake Oehler Morrison <jomorrison@gmail.com>

technerdlove committed:
Clean up formatting of if statement

WordPress coding standards for space usage:  Add space in parens for $title parameter as follows:  if ( $permalink && empty( $title ) )

Co-Authored-By: Jake Oehler Morrison <jomorrison@gmail.com>

technerdlove commented:
Oh, hello! I can work on this starting Monday.
Thanks for letting me know,
Annmarie

On Thu, Jun 2, 2022 at 7:11 AM Onu Abah ***@***.***> wrote:

> Hey there @technerdlove <https://github.com/technerdlove>
> Any chance you could fix merge conflicts before we merge?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Automattic/WP-Job-Manager/pull/1854#issuecomment-1144914198>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADU7RY3AX676HO7FREN6JLTVNC6J3ANCNFSM4IKGEDFA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
--------------------------------------------
Q: Why is this email five sentences or less?
A: http://five.sentenc.es

gikaragia commented:
This was implemented as part of #2262
